dashboard/app/controllers/registrations_controller.rb

```diff
@@ -8,6 +8,8 @@
 require 'queries/lti'
 
 class RegistrationsController < Devise::RegistrationsController
+  before_action :redirect_if_signed_in, only: [:account_type, :login_type, :finish_student_account, :finish_teacher_account]
```

Comment:
Could you use the [Devise-provided](https://github.com/heartcombo/devise/blob/72884642f5700439cc96ac560ee19a44af5a2d45/app/controllers/devise/registrations_controller.rb#L4) `require_no_authentication` here?

Reply:
Definitely! I'm unfamiliar with this repo. I'm reading through its README but can you explain how you knew to look here, or how you know what this action means?

Reply:
Devise is the gem we use for authentication, which is why I looked there. As for knowing what this action means, I'm taking a guess based on the name and behavior we already see in our sign up process. Plus, I figured the test you already wrote would catch an issue if it did something completely strange :)

Reply:
I see it in the passwords controller but am not sure I understand its use here. I might reach out offline!

Reply:
It's implicitly already used in this controller as this controller inherits from `Devise::RegistrationsController`. Currently only applies to `new`, `create` and `cancel` though.

Reply:
I don't know how the string piece works 😅 But yay glad you got it to work!